### ISSUE TITLE
Add Swarm platform support

### ISSUE THREAD

**Issue by crackerplace** [solo-io/squash#11]
Just started getting some comfort level with the code.
Would like to to add swarm support for squash incrementally.

**Comment by ilevine** [comment]
sounds great! please let us know how we can help - squash slack: https://slack.solo.io

**Comment by yuval-k** [comment]
@crackerplace Feel free to lets us know here or on slack about your plans.

as we are starting to form plans ourselves for squash v2.
The internal (debuggers, platforms) apis will probably not change much, but we are trying to make the external  user-facing apis easier for consumption.

One things that we are considering is removing the ServiceWatcher interface, and support it via service mesh instead (currently via istio\envoy). this gives the advantage of only having to support one type of functionality (attach to a container).

Feedback regarding how this works with swarm is welcome.

**Comment by crackerplace** [comment]
@ilevine @yuval-k 

Thanks.I think I got an overall picture of different interfaces.Though I did not work much on swarm in particular but as I am comfortable with docker ,was of the idea that this integration should be a good starting point.Let me know if you have some other points.

I will have few questions might be in a day once I scan the code a bit more.For this [example](https://github.com/solo-io/squash/blob/master/docs/getting-started.md) ,which lists down steps for debugging using a containerid and which looks like a good start to get a first cut working for newer integrations,I believe ServiceWatcher might not be needed ?

On the istio/envoy reference,I did not get a chance to look much into envoy as such.

**Comment by yuval-k** [comment]
these involve a user facing api, which is probably going to change soon.

I'd recommend you take a look in https://github.com/solo-io/squash/blob/master/pkg/platforms/interface.go to see the platform dependent interfaces, that are less likely to change.

Also figure out the how to deploy the squash agent that needs to be present on the nodes - today DaemonSet is used to deploy it in kubernetes.

**Comment by crackerplace** [comment]
yes,I saw that [daemonset](https://raw.githubusercontent.com/solo-io/squash/master/contrib/kubernetes/squash-ds.yml) is used for deploying client to all nodes.

docker swarm 1.12 has global mode,to deploy to all nodes.

**Comment by phantomis** [comment]
Any help adding this functionality?
